pointers/pointers.c

```diff
@@ -19,9 +21,21 @@ void swap(int* a, int* b)
 
     Do not use the `strchr` function from the standard library.
 */
-char *find_char(char *str, int c)
+int string_length(char s[])
 {
+    int i;
+    for(i = 0; s[i] != '\0'; ++i);
+    return i;
+}
 
+char *find_char(char *str, int c)
+{
+    for (int i=0; string_length(str) > i; ++i){
+        if (str[i]==c){
+          return &str[i];
+        }
+    }
+    return NULL;
```

Comment:
This is a very creative solution! There is a more efficient way to do this but it works!